File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        maxWater = 0
        area = 0
        while left < right:
                area = (right - left) * min(height[left], height[right])
                maxWater = max(maxWater, area)

                if height[left] < height[right]:
                    left += 1
                else:
                    right -= 1
        return maxWater

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int result = 0;
        int left = 0;
        int right = height.length - 1;
        while(left < right) {
            int water = (right - left) * Math.min(height[left], height[right]);
            result = Math.max(result, water);

            if(height[left] < height[right]) {
                left += 1;
            } else {
                right -=1;
            }
        }
        return result;
    }
}